src/options/getRepoOptions.ts

```diff
@@ -0,0 +1,40 @@
+import { cosmiconfigSync } from 'cosmiconfig';
+import { getDefaultRemoteBranch } from 'workspace-tools';
+import { RepoOptions, CliOptions } from '../types/BeachballOptions';
+
+export function getRepoOptions(cliOptions: CliOptions): RepoOptions {
+  let repoOptions: RepoOptions | null;
```

Comment:
**suggestion:** nesting ifs adds cognitive load and is hard to understand the program flow in general (also I don't see tests for this)

I'd recommend exit early approach with immutable approach.

Simple example:
> any kind of mutation adds complexity so this example is not ideal - primarily applied exit early strategy and removed nesting

```ts
function getRepoOptions(cliOptions: CliOptions){
  let repoOptions = tryLoadConfig(cliOptions.configPath);

  if(repoOptions === null){
    console.error(`Config file "${cliOptions.configPath}" could not be loaded`);
    process.exit(1);
  }

  repoOptions = trySearchConfig() || {} 

  if (!cliOptions.branch)
    const branch = resolveBranch(repoOptions,cliOptions)
    repoOptions.branch = branch
  }

  return repoOptions
}
```

Reply:
Agreed about the nesting in general but I'm not seeing a reasonable way to further simplify it while still handling all required cases.